pkg/util/util.go

```diff
@@ -81,6 +82,23 @@ func NeedVirtioNetDevice(vmi *v1.VirtualMachineInstance, useEmulation bool) bool
 	return false
 }
 
+// UseSoftwareEmulationForDevice determines whether to fallback to software emulation for the given device.
+// This happens when the given device doesn't exist, and software emulation is enabled.
+func UseSoftwareEmulationForDevice(devicePath string, useEmulation bool) (bool, error) {
```

Comment:
I hope we can rename this arg (now or in a follow-up PR) to something like
```suggestion
func UseSoftwareEmulationForDevice(devicePath string, isEmulationAllowed bool) (bool, error) {
```

Reply:
Will do in a follow-up PR. For now I chose to stick to the existing terminology, particularly since it's used in other functions in this file.

Reply:
See #6201